docs/tutorials/google/advanced_calibration.ipynb

```diff
@@ -0,0 +1,799 @@
+{
```

Comment:
<p>A rather important bug, that I think is prevalent in the loschmidt echos notebook as well: try doing cycles=1. How many two-qubit layers do you see? How about cylces=2?</p><p> </p><p>This is because you are using grid_staggered_pattern but a horizontal line of qubits (instead of a grid). Related but not necessarily sufficient: https://github.com/quantumlib/Cirq/issues/3833</p><p> </p><p>Options:</p><ol><li>have the random generation function ignore the gridinteractionlayers that don't produce interactions. This would solve the issue at hand but wouldn't produce optimal circuits if you had a line of qubits that was snaked on a diagonal or something</li><li>Have a special function or pattern to do random circuits on linear topologies</li><li>Have a be-all-end-all random circuit generation function that works on arbitrary planar topologies by doing edge coloring to figure out activation orders.</li><li>Make cycles always a multiple of 4</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p>Thanks for pointing this out. Discovering this was the main reason the Loschmidt tutorial took so long.</p><p> </p><p>For now I kept cycles as a multiple of 4, but the topology is also no longer linear. </p><p> </p><p>Would you really call this a bug, though? It makes computing the per-two-qubit-gate error harder, but every circuit in the Loschmidt batch has the same number of two-qubit gates, so its valid to compare.</p>

Reply:
It's surprising. I added a variant of the above comment to #3342